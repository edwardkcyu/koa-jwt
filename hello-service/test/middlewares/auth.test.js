require('dotenv').config({ path: 'test/.env.test' });

const { auth } = require('../../src/middlewares');
const HttpError = require('../../src/lib/HttpError');

jest.mock('axios');
const axios = require('axios');

describe('Auth middleware', () => {
  let ctx;

  beforeEach(() => {
    axios.mockResolvedValue({ data: { userName: 'curry' } });
    ctx = { header: { authorization: 'DUMMY' } };
  });

  afterEach(() => {
    axios.mockRestore();
  });

  test('works', async () => {
    await auth(ctx, () => {});

    expect(ctx.auth).toMatchSnapshot();
  });

  test('throws error for missing authorization header', async () => {
    delete ctx.header.authorization;

    try {
      await auth(ctx, () => {});
    } catch (e) {
      expect(e).toBeInstanceOf(HttpError);
    }
  });

  test('throws error when auth service has error', async () => {
    axios.mockImplementation(() => {
      throw new Error('request error');
    });

    try {
      await auth(ctx, () => {});
    } catch (e) {
      expect(e).toBeInstanceOf(HttpError);
    }
  });
});
