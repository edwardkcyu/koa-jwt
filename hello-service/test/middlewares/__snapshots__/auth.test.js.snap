// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Auth middleware works 1`] = `
Object {
  "userName": "curry",
}
`;
